Firestore/Source/Remote/FSTRemoteStore.m

```diff
@@ -540,6 +540,7 @@ - (void)startWriteStream {
 
 - (void)cleanUpWriteStreamState {
   self.lastBatchSeen = kFSTBatchIDUnknown;
+  FSTLog(@"Stopping write stream with %i pending writes", [self.pendingWrites count]);
```

Comment:
This should be producing a warning. Please make sure your setup is showing it to you.

`self.pendingWrites` is an `NSMutableArray` which means `count` is an `NSUInteger`, which isn't suitable for direct use with printf format strings.

I think this means the format should be `%lu` and you need to cast to `unsigned long` before using it. A comparable usage within the code would be https://github.com/firebase/firebase-ios-sdk/blob/13da48d8a4fcdfe288fa15c788c59e7f54edf42c/Firestore/Source/Model/FSTMutationBatch.m#L77